website/docs/reference/widgets/list.md

```diff
@@ -41,70 +35,110 @@ You can display static JSON data in the `Items` property for generating the list
 
 Follow the steps below to bind each JSON field to the widgets embedded in the List:
 
-* Select the Image widget and add  `{{currentItem.bookImage}}` under `Image` in the property pane. `currentItem` refers to the data for a particular item. All the list items get populated based on the corresponding data in the JSON object.
+* Select the Image widget and add  `{{currentItem.bookImage}}` under **Image** property. **currentItem** refers to the data for a particular item. All the list items get populated based on the corresponding data in the JSON object.
 * You can now see the image in the list item, as the image widget renders the image available on the URL supplied in JSON.
 
 Similarly, you can bind `bookName`, `bookId`, and `price` to the embedded widgets in the List widget.
 
 ### Dynamic data mapping
 
-If you want to bind the response from a query or a JS function, then you can use mustache syntax (`{{ }}`). Use the format of `{{QUERY_NAME.data}}` to bind the data returned by the query. For example, You a query `GetAllEmployees`, bind the response in the `Items` property as shown below:
+If you want to bind the response from a query or a JS function, then you can use mustache syntax `{{ }}`. Use the format of `{{QUERY_NAME.data}}` to bind the data returned by the query. For example, You a query `GetAllEmployees`, bind the response in the **Items** property as shown below:
 
 ```javascript
-
 {{GetAllEmployees.data}}
 ```
 
 To learn how to bind data from JS functions, see [Display Data from JS function](/core-concepts/writing-code/workflows#display-data-from-async-js-function)
 
-If you are binding the dynamic data to the List widget, remember to use the  columns/attribute name to map to the individual widget as shown below:
+To display the data in individual widgets in the list item cards, use the **currentItem** property to bind the corresponding value from the object's fields in the widget as shown below. 
  
 ```javascript
-
 {{currentItem.<attribute_or_column_name>}}
 ```
 
+where `currentItem` for the first list item reflects the 0th object in the dataset. This can be used anywhere within a widget that's placed inside the List widget.
+
+
+### Unique list item identifier
+
+The List widget needs to identify each item uniquely to update, reorder, add or remove them. Similar to the `Primary Key` concept in the database or `key` in React, an identifier should be selected from the **Data identifier** property dropdown whose values are unique in the dataset provided to the List widget. Always set the **Data Identifier** property with a valid unique identifier to boost performance.
+
+In the preceding example, the identifier `bookId` has a unique value in the dataset.
+
+If no such unique identifier is present in the data, you can join multiple identifiers  to form a unique pattern by enabling the `JS` mode in the property.
+
+**Example:**
+```
+{{currentItem.bookName + "_" + currentItem.author}}
+```
+
```

Comment:
we should add a warning/info to let people know they must configure this for a smooth list widget experience

Reply:
@dilippitchika what do we mean by a smooth list widget experience. as what could go wrong if they don't set it so that we can explain it a bit more clearly in the callout?